docs: explain how use local NodeSDK library. (#4265)
dagger/dagger: sdk/nodejs/README.md

npm install @dagger.io/dagger --save-dev
```

## Local development

You may want to work on the NodeSDK and test it directly on a local node project. 

1. Create a new Node project

:warning: Skip this step if you already have a node project

```shell
mkdir my-test-ts-project

# Init project (you may use yarn or pnpm)
npm init -y

# Add typescript
npm install typescript --save-dev

# Init typescript project
npx tsc --init
```

2. Update project settings

Dagger exports its SDK using type module so you will need to also update
your `package.json` to the same type.

Add or update the field `type` in your `package.json`

```json
"type": "module"
```

You must also update your `tsconfig.json` to use `NodeNext` as `module`.

```json
"module": "NodeNext"
```

3. Import Dagger local module

Go to the Dagger nodejs SDK directory and do the following :

```shell

> cd path/to/dagger/sdk/nodejs

Reviewer: ```suggestion
cd path/to/dagger/sdk/nodejs
yarn link
```

Reviewer: My previous suggestion has deleted the `yarn link` my bad 🙏

Reviewer 2: Fixed while rebasing his branch. @TomChv I forced pushed on top of your branch, to make this PR advance 😇

npm install @dagger.io/dagger --save-dev
```

## Local development

You may want to work on the NodeSDK and test it directly on a local node project. 

1. Create a new Node project

:warning: Skip this step if you already have a node project

```shell
mkdir my-test-ts-project

# Init project (you may use yarn or pnpm)
npm init -y

# Add typescript
npm install typescript --save-dev

# Init typescript project
npx tsc --init
```

2. Update project settings

Dagger exports its SDK using type module so you will need to also update
your `package.json` to the same type.

Add or update the field `type` in your `package.json`

```json
"type": "module"
```

You must also update your `tsconfig.json` to use `NodeNext` as `module`.

```json
"module": "NodeNext"
```

3. Import Dagger local module

> The simplest part, jump to the Dagger NodeSDK module and tip `yarn link`.

Reviewer: ```suggestion
Go to the Dagger nodejs SDK directory and do the following :
```

> ```shell

Reviewer: ```suggestion
```shell
cd path-to-dagger/sdk/nodejs

> yarn link

Reviewer: ```suggestion

```

> yarn link v1.22.19

Reviewer: ```suggestion

```

> success Registered "@dagger.io/dagger".

Reviewer: ```suggestion

```

> info You can now run `yarn link "@dagger.io/dagger"` in the projects where you want to use this package and it will be used instead.

Reviewer: ```suggestion

```

```

> Go back to your local project and tip `yarn link @dagger.io/dagger`.

Reviewer: ```suggestion
Go back to your local project to link the nodejs sdk directory.
```

```shell
yarn link @dagger.io/dagger  

> yarn link v1.22.19

Reviewer: ```suggestion

```

> success Using linked package for "@dagger.io/dagger".

Reviewer: ```suggestion

```

> ✨  Done in 0.02s.

Reviewer: ```suggestion

```
